feat(platform-browser): Warns on conflicting hydration and blocking n…

SkyZeroZx opened the pull request:
Adds an internal injection token to detect and warn when both hydration and blocking initial navigation are enabled simultaneously.

Since this combination is currently unsupported and may lead to unexpected runtime issues, a warning is now logged during application initialization to alert developers early in the lifecycle.

This helps prevent subtle misconfigurations and aligns with the resolution of https://github.com/angular/angular/issues/59624

Changes included:

- Introduced an internal token to detect the configuration conflict.

- Logged a console.warn during app bootstrap when both features are enabled.

JeanMeche commented on packages/core/src/hydration/tokens.ts:
```diff
@@ -68,3 +68,10 @@ export const JSACTION_BLOCK_ELEMENT_MAP = new InjectionToken<Map<string, Set<Ele
+ * Internal token that indicates whether the initial navigation is blocking in the application.
+ */
+export const IS_ENABLED_BLOCKING_INITIAL_NAVIGATION = new InjectionToken<boolean>(
+  typeof ngDevMode === 'undefined' || !!ngDevMode ? 'IS_ENABLED_BLOCKING_INITIAL_NAVIGATION' : '',
+);
```
nit: This double bang isn't necessary. 
```suggestion
  typeof ngDevMode === 'undefined' || ngDevMode ? 'IS_ENABLED_BLOCKING_INITIAL_NAVIGATION' : '',
);
```

SkyZeroZx replied:
Update it , and similar for other tokens in the file

SkyZeroZx committed:
feat(platform-browser): Warns on conflicting hydration and blocking navigation

Adds an internal token to detect when both hydration and blocking initial navigation are enabled. Logs a warning during app initialization if this unsupported combination is found, helping developers avoid misconfiguration and potential runtime issues.

thePunderWoman reviewed (approved):
LGTM! Thanks!
